src/Workspaces/Core/Portable/SymbolKey/SymbolKey.TupleTypeSymbolKey.cs

```diff
@@ -14,14 +14,17 @@ public static void Create(INamedTypeSymbol symbol, SymbolKeyWriter visitor)
             {
                 Debug.Assert(symbol.IsTupleType);
                 visitor.WriteSymbolKey(symbol.TupleUnderlyingType);
-                visitor.WriteStringArray(symbol.TupleElementNames);
 
+                var frieldlyNames = ArrayBuilder<String>.GetInstance();
                 var locations = ArrayBuilder<Location>.GetInstance();
-                for (var i = 0; i < symbol.TupleElementTypes.Length; i++)
+
+                foreach(var element in symbol.TupleElements)
                 {
-                    locations.Add(symbol.GetMembers("Item" + (i + 1)).FirstOrDefault()?.Locations.FirstOrDefault());
+                    frieldlyNames.Add(element.IsImplicitlyDeclared? null: element.Name);
```

Comment:
Fixed

---
In reply to: [86480535](https://github.com/dotnet/roslyn/pull/14958#discussion_r86480535) [](ancestors = 86480535)